projects/bp-gallery/src/components/common/picture-gallery/PictureTimeline.tsx

```diff
@@ -0,0 +1,107 @@
+import { ArrowDropDown } from '@mui/icons-material';
+import { debounce } from 'lodash';
+import { Dispatch, SetStateAction, useCallback, useEffect, useMemo, useRef } from 'react';
+import { useAnimate } from '../../../hooks/animate.hook';
+
+export const enum TimeStepType {
+  DECADE = 'decade',
+  YEAR = 'year',
+  MONTH = 'month',
+  DAY = 'day',
+}
+
+const TEXT_OVERFLOW_WIDTH = 20;
+
+const PictureTimeline = ({
+  start,
+  end,
+  type,
+  date,
+  setDate,
+  singleElementWidth = 80,
+}: {
+  start: number;
+  end: number;
+  type: TimeStepType;
+  date: number;
+  setDate: Dispatch<SetStateAction<number>>;
+  singleElementWidth?: number;
+}) => {
+  const scrollBarRef = useRef<HTMLDivElement>(null);
+  const [scrollLeft, scrollTo] = useAnimate(
+    (date - start) * singleElementWidth + singleElementWidth / 2 + TEXT_OVERFLOW_WIDTH,
+    0.06
+  );
+
+  useEffect(() => {
+    if (scrollBarRef.current) {
+      scrollBarRef.current.scrollLeft = scrollLeft;
+    }
+  }, [scrollBarRef, scrollLeft]);
+
+  const updateDate = useCallback(() => {
+    if (scrollBarRef.current) {
+      scrollTo(scrollBarRef.current.scrollLeft);
+      const year =
+        Math.floor(
+          (scrollBarRef.current.scrollLeft + singleElementWidth / 2 + TEXT_OVERFLOW_WIDTH) /
+            singleElementWidth
+        ) +
+        start -
+        1;
+      setDate(year);
+    }
+  }, [scrollTo, setDate, singleElementWidth, start]);
+
+  const listItems = [];
+  for (let year = start; year <= end; year++) {
+    listItems.push(
+      <li
+        key={year}
+        className='inline cursor-pointer'
+        onClick={() => {
+          setDate(year);
+        }}
+        style={{
+          padding: `40px ${
+            singleElementWidth - TEXT_OVERFLOW_WIDTH
+          }px 20px ${TEXT_OVERFLOW_WIDTH}px`,
+        }}
+      >
+        <div className='relative inline w-full bottom-0'>
+          <hr
+            className={`absolute left-0 ${
+              date === year || date === year - 1 ? '-top-1' : 'top-0'
+            } ${date === year || date === year - 1 ? 'h-[20px]' : 'h-[16px]'}`}
+          />
+          <span
+            className={`absolute -left-[18px] ${
+              date === year || date === year - 1 ? '-top-5' : '-top-4'
+            } select-none ${date === year || date === year - 1 ? 'text-black' : 'text-slate-500'}`}
```

Comment:
Maybe extract `date === year || date === year - 1` into a `const isHighlighted` at the top of the loop body.
Also, instead of having two ternaries, you could have one each, i. e. `isHighlighted ? '-top-1 h-[20px]' : 'top-0 h-[16px]'`.